dbms/src/Storages/DeltaMerge/Segment.cpp

```diff
@@ -1387,7 +1388,19 @@ Segment::prepareSplitLogical( //
     {
         auto ori_page_id = dmfile->pageId();
         auto file_id = dmfile->fileId();
-        auto file_parent_path = delegate.getDTFilePath(file_id);
+        String file_parent_path;
+        if (dm_context.db_context.getSharedContextDisagg()->remote_data_store)
+        {
+            auto wn_ps = dm_context.db_context.getWriteNodePageStorage();
+            auto full_page_id = UniversalPageIdFormat::toFullPageId(UniversalPageIdFormat::toFullPrefix(StorageType::Data, dm_context.storage_pool->getNamespaceId()), ori_page_id);
+            auto remote_data_location = wn_ps->getCheckpointLocation(full_page_id);
+            const auto & lock_key_view = S3::S3FilenameView::fromKey(*(remote_data_location->data_file_id));
+            file_parent_path = S3::S3Filename::fromTableID(lock_key_view.store_id, dm_context.storage_pool->getNamespaceId()).toFullKeyWithPrefix();
+        }
```

Comment:
Can we get the parent_path from the existing `dmfile` object directly?

Reply:
https://github.com/pingcap/tiflash/blob/4a204ea07139a1b2cff2028f1190c923e9acd8b0/dbms/src/Storages/DeltaMerge/File/DMFile.h#L458-L463

Reply:
Fixed.